articles/modules/ROOT/pages/protecting-against-SSRF.adoc

```diff
@@ -0,0 +1,119 @@
+
+
+The use case "users should be able to load the data from anywhere on the internet" is falling into https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html#case-2-application-can-send-requests-to-any-external-ip-address-or-domain-name[Case 2^], which is more complex to defend against.
+
+But with an _defence in depth approach_, it is possible.
+
+Neo4j does input validation as much as possible, which is your *application layer* defence. And if you build custom applications on top of neo4j, it's a good idea to limit the ability to upload data and to run raw cypher queries to your authenticated users.
+But to ensure https://csrc.nist.gov/glossary/term/defense_in_depth[_defence in depth_^], you should do more on the *network layer*.  
+
+The detection and prevention are highly specific to your infrastructure and network environment.
+
+=== Running on AWS cloud
+
+Use https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/[IMDSv2^].
+
+=== Running on Google cloud
+
+Google has advisories for preventing access to https://cloud.google.com/compute/docs/metadata/querying-metadata[metadata^], https://cloud.google.com/compute/docs/instances/protecting-resources-vpc-service-controls[Compute Engine resources^] and how to https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity[utilize workload identity to minimize access^].
+
+=== Generic mitigation
+
+Segment your network and configure your firewall to prevent access from Neo4j to the metadata subnet (if running on cloud) and any other sensitive backend subnets.
+In most cases you want to prevent access to internal IP addresses originating from the backend service in general.
+
+== Configuration option
+
+In addition to the external network-level restriction mechanisms listed above, Neo4j has an internal control mechanism which allows users to provide IP address ranges for the DBMS to block.
+This control is activated via a configuration setting which must be added to https://neo4j.com/docs/operations-manual/current/configuration/neo4j-conf[neo4j.conf^] and
+requires restarting the DBMS to be applied.
+
+=== `unsupported.dbms.cypher_ip_blocklist`
+[cols="1,5"]
+|===
+| **Description** | IP address ranges used by the DBMS to block IP requests
+| **Valid values** | Comma separated list of IPv4 and IPv6 https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing#CIDR_notation[CIDR-notation^] IP ranges
+| **Default value** |
+|===
+
+When the DBMS attempts to make an IP request to a URL, after resolving the URL to an IP address, it ensures this address doesn't collide with any
+of the blocked IP ranges provided by the configuration setting. Because the control mechanism is applied at the IP level, it will be applied to any HTTP and FTP requests made by Cypher features
```

Comment:
```suggestion
of the blocked IP ranges provided by the configuration setting. Because the control mechanism is applied at the IP network level, it will be applied to any HTTP and FTP requests made by Cypher features
```